### ISSUE TITLE
x-csrf-token callback does not process the response

### ISSUE THREAD

**Issue by cuhland** [SAP/openui5-fhir#85]
**Describe the bug**
With this setting openui5-fhir is supposed to fetch a token and use that token for further requests.
```
"type": "sap.fhir.model.r4.FHIRModel",
"settings": {
    "x-csrf-token": true
}
```

But when I configure this then the following error occurs:
```
> Uncaught TypeError: Cannot read property 'getAllResponseHeaders' of undefined
    at FHIRRequestor.getResponseHeaders (FHIRRequestor.js:497)
    at FHIRRequestor._callBackForXcsrfToken (FHIRRequestor.js:268)
    at FHIRRequestor.<anonymous> (FHIRRequestor.js:308)
```

This is the IMHO the code in the FHIRRequestor.js which causes the problem
```javascript
if (this.bCSRF && !this.sToken){
  mHeaders["x-csrf-token"] = "fetch";
  var tmpFnSuccess = FHIRUtils.deepClone(fnSuccess);
  var tmpFnError = FHIRUtils.deepClone(fnError);
  fnSuccess = this._callBackForXcsrfToken.bind(this, tmpFnSuccess);
  fnError = function(oRequestHandle, oData){
    tmpFnError(oRequestHandle, oData);
  };
  return fnTriggerRequest();
```

**Expected behavior**
As suggested in #68 the `_callBackForXcsrfToken` should use the `oRequestHandle` to set `this.sToken`. The request was triggered with the the header to fetch the token, so it should be set.

```
this.sToken = this.getResponseHeaders(oRequestHandle._jqRequest)["x-csrf-token"];
```

**OpenUI5-FHIR version**
- *OpenUI5-FHIR* 1.0.4

**OpenUI5 version**
- *OpenUI5* 1.64.3

**Further information**
- See this branch in repo for an example https://github.com/cuhland/openui5-fhir-sample-app/tree/cuh/demo-fixTokenFetch

**Comment by flovogt** [comment]
Hi @cuhland, sorry for this long ongoing issue. We've checked your issue and we'll provide a fix for that.

**Comment by flovogt** [comment]
PR https://github.com/SAP/openui5-fhir/pull/108/ fixes the issue

**Comment by flovogt** [comment]
Hi @cuhland ,
the fix is implemented with release [2.0.3](https://github.com/SAP/openui5-fhir/releases/tag/v2.0.3).
Thanks a lot for opening this issue.
Best Regards,
Florian
